primitives/teerex/src/lib.rs

```diff
@@ -35,31 +35,110 @@ impl Default for SgxBuildMode {
 }
 
 #[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, sp_core::RuntimeDebug, TypeInfo)]
-pub enum AttestationMethod {
-	Dcap,
+pub enum SgxAttestationMethod {
+	Skip(bool),
 	Ias,
-	Skip,
+	Dcap(bool),
+}
+
+impl Default for SgxAttestationMethod {
+	fn default() -> Self {
+		SgxAttestationMethod::Skip(false)
+	}
+}
+
+const SGX_REPORT_DATA_SIZE: usize = 64;
+#[derive(Debug, Encode, Decode, Copy, Clone, PartialEq, Eq, TypeInfo)]
+#[repr(C)]
+pub struct SgxReportData {
+	pub d: [u8; SGX_REPORT_DATA_SIZE],
+}
+
+impl Default for SgxReportData {
+	fn default() -> Self {
+		SgxReportData { d: [0u8; SGX_REPORT_DATA_SIZE] }
+	}
+}
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, sp_core::RuntimeDebug, TypeInfo)]
+pub enum SgxStatus {
+	Invalid,
+	Ok,
+	GroupOutOfDate,
+	GroupRevoked,
+	ConfigurationNeeded,
+}
+impl Default for SgxStatus {
+	fn default() -> Self {
+		SgxStatus::Invalid
+	}
 }
```

Comment:
Nowadays, this can be replaced with:

```
pub enum SgxStatus {
        #[default]
	Invalid,
	Ok,
	GroupOutOfDate,
	GroupRevoked,
	ConfigurationNeeded,
}
```